tests/renderer/state-spec.ts

```diff
@@ -560,4 +560,11 @@ describe('AppState', () => {
       });
     });
   });
+
+  describe('resetEditorLayout()', () => {
+    it('Puts editors in default arrangement', () => {
+      appState.resetEditorLayout();
+      expect(appState.mosaicArrangement).toEqual(DEFAULT_MOSAIC_ARRANGEMENT);
```

Comment:
It's a good test but then not very exponential. State by default generates mosaic with default arrangement and tests can pass when nothing is changed.

Here most optimal it's changing the default arrangement of mosaic for more credibility that test passed when something is really changed.

Here example for your case:
```suggestion
      appState.hideAndBackupMosaic(EditorId.main);

      expect(appState.mosaicArrangement).toEqual({
        direction: 'row',
        first: EditorId.renderer,
        second: EditorId.html
      });

      appState.resetEditorLayout();
      expect(appState.mosaicArrangement).toEqual(DEFAULT_MOSAIC_ARRANGEMENT);
      expect(appState.mosaicArrangement).toEqual({
        direction: 'row',
        first: EditorId.main,
        second: {
          direction: 'column',
          first: EditorId.renderer,
          second: EditorId.html,
        }
      });
```

Here we remove `main` window and verify that the only `renderer` and `html` are visible, and at the end, we check if all reset by default.

Reply:
Good point, it'd be better to test it this way since we indeed verify that the state changes. However, I would say that the last `expect()` call is redundant here since `{
        direction: 'row',
        first: EditorId.main,
        second: {
          direction: 'column',
          first: EditorId.renderer,
          second: EditorId.html,
        }
      }` is already hard-coded to `DEFAULT_MOSAIC_ARRANGEMENT`.

Reply:
Great flag, thank you so much @HashimotoYT! You and @erickzhao are helping me learn a lot about testing, and I updated the test based on your comments.